include/libdnf/base/goal.hpp

```diff
@@ -54,53 +54,285 @@ class Goal {
+    // @replaces libdnf/hy-goal.h:function:hy_goal_install(HyGoal goal, DnfPackage *new_pkg)
+    // @replaces libdnf/hy-goal.h:function:hy_goal_install_optional(HyGoal goal, DnfPackage *new_pkg)
     void add_rpm_install_or_reinstall(
         const libdnf::rpm::Package & rpm_package, const libdnf::GoalJobSettings & settings);
+
+    /// Add install_or_reinstall job to the goal. In default it uses `clean_requirements_on_remove` with `false` value.
+    ///
+    /// @param package_set      A package_set to be added to the install_or_reinstall job.
+    // @replaces libdnf/hy-goal.h:function:hy_goal_install_selector(HyGoal goal, HySelector sltr, GError **error)
+    // @replaces libdnf/hy-goal.h:function:hy_goal_install_selector_optional(HyGoal goal, HySelector sltr, GError **error)
     void add_rpm_install_or_reinstall(const libdnf::rpm::PackageSet & package_set);
+
+    /// Add install_or_reinstall job to the goal. In default it uses `clean_requirements_on_remove` with `false` value.
+    ///
+    /// @param package_set      A package_set to be added to the install_or_reinstall job.
+    /// @param settings         A structure to override default goal settings. Only `strict`, `best`, and `clean_requirements_on_remove` are used.
+    // @replaces libdnf/hy-goal.h:function:hy_goal_install_selector(HyGoal goal, HySelector sltr, GError **error)
+    // @replaces libdnf/hy-goal.h:function:hy_goal_install_selector_optional(HyGoal goal, HySelector sltr, GError **error)
     void add_rpm_install_or_reinstall(
         const libdnf::rpm::PackageSet & package_set, const libdnf::GoalJobSettings & settings);
+
+    /// Add reinstall job to the goal for future resolvement. In default it uses `clean_requirements_on_remove` with `false` value.
+    ///
+    /// @param spec      A string describing the Goal reinstall job.
+    // @replaces dnf:dnf/base.py:method:Base().reinstall(self, pkg_spec, old_reponame=None, new_reponame=None, new_reponame_neq=None, remove_na=False)
     void add_rpm_reinstall(const std::string & spec);
+
+    /// Add reinstall job to the goal for future resolvement. In default it uses `clean_requirements_on_remove` with `false` value,
+    /// but it can be overridden by setting.
+    ///
+    /// @param spec      A string describing the Goal reinstall job.
+    /// @param settings  A structure to override default goal settings. The value `from_repo_ids` is not used (not implemented yet).
+    // @replaces dnf:dnf/base.py:method:Base().reinstall(self, pkg_spec, old_reponame=None, new_reponame=None, new_reponame_neq=None, remove_na=False)
     void add_rpm_reinstall(const std::string & spec, const libdnf::GoalJobSettings & settings);
+
     // TODO(jmracek) Do we want to add reinstall or remove?
+
+    /// Add remove job to the goal for future resolvement. In default it uses `clean_requirements_on_remove` according to Config.
```

Comment:
What do you mean by _"In default it uses `clean_requirements_on_remove` according to Config."_? The dnf config file? Needs clarification in the docstring and it seems this is a weird inconsistency (with sometimes the value being hardcoded `false`) that may need some special handling in the documentation.

Reply:
`clean_requirements_on_remove` is configuration option. Some methods use the value from the config (remove methods). But other methods like upgrade use as default value for `clean_requirements_on_remove` false. Then the value can be overridden by settings.